src/main/java/org/jabref/gui/preferences/BibtexKeyPatternTabViewModel.java

```diff
@@ -16,46 +16,51 @@
-    private StringProperty keyPatternReplacementProperty = new SimpleStringProperty();
-    private StringProperty unwantedCharactersProperty = new SimpleStringProperty();
+    private final BooleanProperty overwriteAllowProperty = new SimpleBooleanProperty();
+    private final BooleanProperty overwriteWarningProperty = new SimpleBooleanProperty();
+    private final BooleanProperty generateOnSaveProperty = new SimpleBooleanProperty();
+    private final BooleanProperty letterStartAProperty = new SimpleBooleanProperty();
+    private final BooleanProperty letterStartBProperty = new SimpleBooleanProperty();
+    private final BooleanProperty letterAlwaysAddProperty = new SimpleBooleanProperty();
+    private final StringProperty keyPatternRegexProperty = new SimpleStringProperty();
+    private final StringProperty keyPatternReplacementProperty = new SimpleStringProperty();
+    private final StringProperty unwantedCharactersProperty = new SimpleStringProperty();
 
     // The list and the default properties are being overwritten by the bound properties of the tableView, but to
     // prevent an NPE on storing the preferences before lazy-loading of the setValues, they need to be initialized.
-    private ListProperty<BibtexKeyPatternPanelItemModel> patternListProperty = new SimpleListProperty<>(FXCollections.observableArrayList());
-    private ObjectProperty<BibtexKeyPatternPanelItemModel> defaultKeyPatternProperty = new SimpleObjectProperty<>(
+    private final ListProperty<BibtexKeyPatternPanelItemModel> patternListProperty = new SimpleListProperty<>(FXCollections.observableArrayList());
+    private final ObjectProperty<BibtexKeyPatternPanelItemModel> defaultKeyPatternProperty = new SimpleObjectProperty<>(
             new BibtexKeyPatternPanelItemModel(new BibtexKeyPatternPanelViewModel.DefaultEntryType(), ""));
 
     private final DialogService dialogService;
     private final JabRefPreferences preferences;
+    private final BibtexKeyPatternPreferences initialBibtexKeyPatternPreferences;
 
     public BibtexKeyPatternTabViewModel(DialogService dialogService, JabRefPreferences preferences) {
         this.dialogService = dialogService;
         this.preferences = preferences;
+        this.initialBibtexKeyPatternPreferences = preferences.getBibtexKeyPatternPreferences();
     }
 
     @Override
     public void setValues() {
-        overwriteAllowProperty.setValue(!preferences.getBoolean(JabRefPreferences.AVOID_OVERWRITING_KEY));
-        overwriteWarningProperty.setValue(preferences.getBoolean(JabRefPreferences.WARN_BEFORE_OVERWRITING_KEY));
-        generateOnSaveProperty.setValue(preferences.getBoolean(JabRefPreferences.GENERATE_KEYS_BEFORE_SAVING));
+        overwriteAllowProperty.setValue(!initialBibtexKeyPatternPreferences.avoidOverwritingCiteKey());
+        overwriteWarningProperty.setValue(initialBibtexKeyPatternPreferences.isWarningBeforeOverwrite());
```

Comment:
I think the `should` prefix is better here: `shouldAvoidOverwriting`, `shouldWarnBeforeOverwrite` and `shouldGenerateKeysBeforeSaving`.
https://stackoverflow.com/a/5326487/873661